ci: Upgrade docker on Ubuntu 20.04 for tests.

mrobertseidowsky-gpsw opened the pull request:
- As reported when I migrated from Ubuntu 19.0 to Ubuntu 20.04, **wheel** was needed into `pynodegl/requirements.txt`.

- Also, **tzdata** installation is needed before **build-essential** to avoid an interactiv dialog blocker.

- This branch has been tested OK on my forked repository: 
[Test on Ubuntu 20.04](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/731136538?check_suite_focus=true)

cboesch-gpsw commented:
I have several questions:
- why installing `tzdata` disable the interactive prompt for `build-essential`?
- if you need to disable interaction, this looks like a better idea to me: https://serverfault.com/questions/227190/how-do-i-ask-apt-get-to-skip-any-interactive-post-install-configuration-steps
- why is `wheel` required? (what's the error?)
- can't we just drop docker? we were using it because 19.10 wasn't available, now you can probably simply use `runs-on: ubuntu-20.04`, no?

mrobertseidowsky-gpsw commented:
> I have several questions:
> 
> * why installing `tzdata` disable the interactive prompt for `build-essential`?
> * if you need to disable interaction, this looks like a better idea to me: https://serverfault.com/questions/227190/how-do-i-ask-apt-get-to-skip-any-interactive-post-install-configuration-steps

`build-essential` stays stuck on :

```
Setting up tzdata (2020a-0ubuntu0.20.04) ...
debconf: unable to initialize frontend: Dialog
debconf: (TERM is not set, so the dialog frontend is not usable.)
debconf: falling back to frontend: Readline
Configuring tzdata
------------------

Please select the geographic area in which you live. Subsequent configuration
questions will narrow this down by presenting a list of cities, representing
the time zones in which they are loc ated.

  1. Africa      4. Australia  7. Atlantic  10. Pacific  13. Etc
  2. America     5. Arctic     8. Europe    11. SystemV
  3. Antarctica  6. Asia       9. Indian    12. US
Geographic area: 
```

So installing tzdata before fixes the "wait for a Geographic area"

I already tried `DEBIAN_FRONTEND=noninteractive apt -y install build-essential` [without success](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/731089119?check_suite_focus=true)


> * why is `wheel` required? (what's the error?)

If I don't specified wheel, I've got [this error](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/731087283?check_suite_focus=true) in the build part: 

```
      Complete output (6 lines):
      usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
         or: setup.py --help [cmd1 cmd2 ...]
         or: setup.py --help-commands
         or: setup.py cmd --help

      error: invalid command 'bdist_wheel'
```

> * can't we just drop docker? we were using it because 19.10 wasn't available, now you can probably simply use `runs-on: ubuntu-20.04`, no?

I already tested and it [doesn't work](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/731035048?check_suite_focus=true) du to the fact that every command must be in "sudo" and `export CFLAGS=-fPIE` is not taken into account.

cboesch-gpsw commented:
> I already tested and it [doesn't work](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/731035048?check_suite_focus=true) du to the fact that every command must be in "sudo" and `export CFLAGS=-fPIE` is not taken into account. I'm gonna test something like `sudo export CFLAGS=-fPIE` make ...` and keep you updated.

You should not need `sudo` to build and install node.gl (everything is deployed in a virtualenv), you only need it for the package installs. We don't use `sudo` currently, what were you trying to fix?

mrobertseidowsky-gpsw commented:
> > I already tested and it [doesn't work](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/731035048?check_suite_focus=true) du to the fact that every command must be in "sudo" and `export CFLAGS=-fPIE` is not taken into account. I'm gonna test something like `sudo export CFLAGS=-fPIE` make ...` and keep you updated.
> 
> You should not need `sudo` to build and install node.gl (everything is deployed in a virtualenv), you only need it for the package installs. We don't use `sudo` currently, what were you trying to fix?

Oh okay, Fixed and [worked](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/733799382?check_suite_focus=true)

cboesch-gpsw commented:
> > > I already tested and it [doesn't work](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/731035048?check_suite_focus=true) du to the fact that every command must be in "sudo" and `export CFLAGS=-fPIE` is not taken into account. I'm gonna test something like `sudo export CFLAGS=-fPIE` make ...` and keep you updated.
> > 
> > 
> > You should not need `sudo` to build and install node.gl (everything is deployed in a virtualenv), you only need it for the package installs. We don't use `sudo` currently, what were you trying to fix?
> 
> Oh okay, Fixed and [worked](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/733799382?check_suite_focus=true)

Good :)

Now I see several interesting things in the log:

> W: --force-yes is deprecated, use one of the options starting with --allow instead.

And also:

> tzdata is already the newest version (2020a-0ubuntu0.20.04).
> build-essential is already the newest version (12.8ubuntu1).
> curl is already the newest version (7.68.0-1ubuntu2).
> pkg-config is already the newest version (0.29.1-0ubuntu4).
> python3-dev is already the newest version (3.8.2-0ubuntu2).
> python3-pip is already the newest version (20.0.2-5ubuntu1).

Maybe you can you skip these? Can you also retry without the `wheel`? Now that you have a clean image and not a random docker, it may help.

mrobertseidowsky-gpsw commented:
> > > > I already tested and it [doesn't work](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/731035048?check_suite_focus=true) du to the fact that every command must be in "sudo" and `export CFLAGS=-fPIE` is not taken into account. I'm gonna test something like `sudo export CFLAGS=-fPIE` make ...` and keep you updated.
> > > 
> > > 
> > > You should not need `sudo` to build and install node.gl (everything is deployed in a virtualenv), you only need it for the package installs. We don't use `sudo` currently, what were you trying to fix?
> > 
> > 
> > Oh okay, Fixed and [worked](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/733799382?check_suite_focus=true)
> 
> Good :)
> 
> Now I see several interesting things in the log:
> 
> > W: --force-yes is deprecated, use one of the options starting with --allow instead.
 
I deleted the `force-yes` since only `y` is needed now.
FYI the recent equivalent of `force-yes` is `--allow-downgrades --allow-remove-essential --allow-change-held-packages` quite a bit long !

> And also:
> 
> > tzdata is already the newest version (2020a-0ubuntu0.20.04).
> > build-essential is already the newest version (12.8ubuntu1).
> > curl is already the newest version (7.68.0-1ubuntu2).
> > pkg-config is already the newest version (0.29.1-0ubuntu4).
> > python3-dev is already the newest version (3.8.2-0ubuntu2).
> > python3-pip is already the newest version (20.0.2-5ubuntu1).
> 
> Maybe you can you skip these? Can you also retry without the `wheel`? Now that you have a clean image and not a random docker, it may help.

- A bit radical but I deleted these package since already there with the `update`: nore more warnings `X is already the newest version` and [test is OK](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/734610807?check_suite_focus=true)

- As reported when I did the migration on my computer from 19 to 20 (no docker), wheel is required, I tested again and the [test is KO](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/actions/runs/123494878/workflow) => is that not normal to require it ?

cboesch-gpsw commented:
> * As reported when I did the migration on my computer from 19 to 20 (no docker), wheel is required, I tested again and the [test is KO](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/actions/runs/123494878/workflow) => is that not normal to require it ?

This is strange, we don't need it here. Somehow on Ubuntu it appears to be looking for a wheel binary of `pynodegl` while it shouldn't (the module is not present so why even try...). Can you test if the following helps instead?

```diff
diff --git a/pynodegl-utils/requirements.txt b/pynodegl-utils/requirements.txt
index d1f89444..c08df859 100644
--- a/pynodegl-utils/requirements.txt
+++ b/pynodegl-utils/requirements.txt
@@ -1,5 +1,5 @@
 pillow
-pynodegl
+pynodegl --no-binary=pynodegl
 # Workaround a regression introduced by pyside2 5.14.2 which makes the node.gl
 # viewer deadlock in various place.
 pyside2==5.14.1
```

mrobertseidowsky-gpsw commented:
> > * As reported when I did the migration on my computer from 19 to 20 (no docker), wheel is required, I tested again and the [test is KO](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/actions/runs/123494878/workflow) => is that not normal to require it ?
> 
> This is strange, we don't need it here. Somehow on Ubuntu it appears to be looking for a wheel binary of `pynodegl` while it shouldn't (the module is not present so why even try...). Can you test if the following helps instead?
> 
> ```diff
> diff --git a/pynodegl-utils/requirements.txt b/pynodegl-utils/requirements.txt
> index d1f89444..c08df859 100644
> --- a/pynodegl-utils/requirements.txt
> +++ b/pynodegl-utils/requirements.txt
> @@ -1,5 +1,5 @@
>  pillow
> -pynodegl
> +pynodegl --no-binary=pynodegl
>  # Workaround a regression introduced by pyside2 5.14.2 which makes the node.gl
>  # viewer deadlock in various place.
>  pyside2==5.14.1
> ```

Despite the option on pynodegl and without wheel: tests is [KO](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/734728077?check_suite_focus=true) (still required wheel)

cboesch-gpsw commented:
> Despite the option on pynodegl and without wheel: tests is [KO](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/734728077?check_suite_focus=true) (still required wheel)

OK well whatever, I won't debug this because it's probably some pip/setuptools or ubuntu bullshit; but then can you add that `wheel` in a separate commit with a comment above? Something along the lines "This is a workaround for the following issue on Ubuntu: `error: invalid command 'bdist_wheel'`"?

mrobertseidowsky-gpsw commented:
> > Despite the option on pynodegl and without wheel: tests is [KO](https://github.com/mrobertseidowsky-gpsw/gopro-lib-node.gl/runs/734728077?check_suite_focus=true) (still required wheel)
> 
> OK well whatever, I won't debug this because it's probably some pip/setuptools or ubuntu bullshit; but then can you add that `wheel` in a separate commit with a comment above? Something along the lines "This is a workaround for the following issue on Ubuntu: `error: invalid command 'bdist_wheel'`"?

Done!

cboesch-gpsw commented:
OK cool, we should be good to go, can you just fix the commit message of the 2nd commit?
- we're not upgrading docker anymore
- the wheel comment is not relevant to this commit anymore
- small cosmetic: can you put the first character after `ci: ` in lower-case?

mbouron-gpsw commented:
LGTM.

mrobertseidowsky-gpsw commented:
> OK cool, we should be good to go, can you just fix the commit message of the 2nd commit?
> 
> * we're not upgrading docker anymore
> * the wheel comment is not relevant to this commit anymore
> * small cosmetic: can you put the first character after `ci: ` in lower-case?

yeap, done!

cboesch-gpsw commented:
I took the liberty to adjust the commit messages a bit more, and it's now merged. Thanks!
